src/App.tsx

```diff
@@ -1,19 +1,113 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import {
+  useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { getTodos, addTodo, deleteTodo } from './api/todos';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { FilterType } from './types/FilterType';
+import { NewTodoField } from './components/NewTodoField';
+import { TodoItem } from './components/TodoItem';
+
+const defaultTodo = {
+  id: 0,
+  userId: 0,
+  title: '',
+  completed: false,
+};
 
-export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const App = (): JSX.Element | null => {
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState(FilterType.All);
+  const [isAdding, setIsAdding] = useState(false);
+  const [temporaryTodo, setTemporaryTodo] = useState(defaultTodo);
+  const [completedIsRemoving, setCompletedIsRemoving] = useState(false);
+  const numberOfTodos = useMemo(() => {
+    return todos.filter((todo) => !todo.completed).length;
+  }, [todos]);
+  const numberOfCompleted = useMemo<number>(() => {
+    return [...todos].filter((todo) => todo.completed).length;
+  }, [todos]);
+  const visibleTodos = useMemo<Todo[]>(() => {
+    if (todos) {
```

Comment:
great) it's amazing that you use memoized variables. Leave one empty line between them to improve code readability)